### DISCUSSION TITLE
Possible to add UPNP

### DISCUSSION THREAD

**Discussion by Redtigercod4** [Haivision/srt#2261] [Ideas]
Looking to use SRT within OBS more so we can stream over the internet when doing events. Issue we have is opening ports up on the router is a no go for one of our locations. I was wondering if it is possible to add upnp feature to SRT to allow us to stream over the internet from that location?

**Comment by eric** [comment]
I'm not certain it would be beneficial to have UPnP integrated directly into libsrt vs having it be done in the application that is using libsrt. The way UPnP would be configured, status displayed, etc, would need to be handled outside of libsrt.

**Reply by marazmarci** [reply]
@Redtigercod4 you can run `srt-live-transmit` on a server somewhere near where you can setup port forwarding.
for example:
```
srt-live-transmit "srt://0.0.0.0:1111?mode=listener&latency=1000" "srt://0.0.0.0:2222?mode=listener&latency=1000"
```
where `1111` is the input port and `2222` is the output port. You can connect to both with an SRT client with the `mode=caller` flag. And of course, you can fine-tune the input and the output's latency. The total latency will be the sum of the input and the output.

For more info, see the docs of `srt-live-transmit`: https://github.com/Haivision/srt/blob/master/docs/apps/srt-live-transmit.md
